### PR TITLE
Mark TreeView as Obsolete

### PR DESCRIPTION
**Pull request by cra0zy**

### CODE REVIEW

File: Source/Eto/Forms/Controls/TreeView.cs [hunk 1]
```diff
@@ -95,6 +95,7 @@ public TreeViewItemEditEventArgs(ITreeItem item, string label)
 	/// </remarks>
 	/// <copyright>(c) 2014 by Curtis Wensley</copyright>
 	/// <license type="BSD-3">See LICENSE for full terms</license>
+	[Obsolete("TreeView is deprecated, please use TreeGridView instead.")]
```
Comment (cwensley) [inline Source/Eto/Forms/Controls/TreeView.cs]: Add "Since 2.3. " to the beginning of that.  Sorry, forgot to mention before. (;

Comment (cwensley) [inline Source/Eto/Forms/Controls/TreeView.cs]: Err. 2.4

**Comment by cra0zy** [comment]
@cwensley Because warnings as errors is enabled I have to disable some stuff with perma rules, I'll do it tomorrow.

**Comment by cwensley** [comment]
I believe if you add `[Obsolete]` to the handler classes it would make it (fairly) easy.

**Comment by cra0zy** [comment]
> I believe if you add [Obsolete] to the handler classes it would make it (fairly) easy.

You would than have to disable the rule in the Handler setup...

**Pull request merged**